### PR TITLE
WIP: Refactor code to use the pre defined image

### PR DESCRIPTION
**Pull request by pkenchap**
We are refactoring the code to use the pre-installed image and removing the pre-installed packages and ocp cli installer.

### CODE REVIEW

**Comment by pkenchap** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.15-ocp-e2e-powervs-ovn-multi-ppc64le

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-ref.yaml [hunk 1]
```diff
@@ -16,10 +16,11 @@ ref:
     default: ""
     documentation: |-
       The terraform version that is installed in support of the installation.
+  - name: OCP_VERSION
+    default: "4.15"
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-ref.yaml:20]:
```suggestion
```

I don't think we need this one

Comment (prb112) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-ref.yaml:20]: It'll be read from file.

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh [hunk 2]
```diff
@@ -59,38 +54,15 @@ function retry {
...
-
-  retry "ic config --check-version=false"
-  retry "ic version"
-}
-
 function setup_terraform_cli() {
-  if [ -z "$(command -v terraform)" ]
-  then
-    echo "terraform CLI doesn't exist, installing"
-  fi
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:85]: We always want to pickup terraform (the version in the container is v1.0.0 or so, not v1.5.5) so the if wasn't providing anything useful.

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh [hunk 3]
```diff
@@ -59,38 +54,15 @@ function retry {
...
   curl -L "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip" -o "${IBMCLOUD_HOME_FOLDER}"/terraform.zip
   cd "${IBMCLOUD_HOME_FOLDER}" || true
   unzip -o "${IBMCLOUD_HOME_FOLDER}"/terraform.zip
   ${IBMCLOUD_HOME_FOLDER}/terraform version
 }
 
 function cleanup_ibmcloud_vpc() {
-  cos_name="${NAME_PREFIX}-mac-intel-cos"
+  cos_name="${NAME_PREFIX}-multi-arch-intel-cos"
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:65]:
```suggestion
  cos_name="${NAME_PREFIX}-multi-arch-intel-cos"
```
We'd have to update our automation to use -multi-arch- https://github.com/IBM/ocp4-upi-compute-powervs-ibmcloud/blob/ba3a92b156fc3c3c93d60e6aca761e6621e4bb6d/modules/5_image/image.tf#L18

That should be a quick update. If you want to create an issue on our internal openshiftp jira project, we can do it quickly.

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh [hunk 4]
```diff
@@ -119,10 +91,10 @@ function cleanup_ibmcloud_vpc() {
 function setup_mac_vpc_workspace(){
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:91]:
```suggestion
function setup_multi_arch_vpc_workspace(){
```

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh [hunk 5]
```diff
@@ -168,8 +140,6 @@ PATH=${PATH}:/tmp
 mkdir -p "${IBMCLOUD_HOME_FOLDER}"
 export PATH=$PATH:/tmp:/"${IBMCLOUD_HOME_FOLDER}"
 
-setup_jq
-setup_ibmcloud_cli
 setup_terraform_cli
 setup_mac_vpc_workspace
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:144]:
```suggestion
setup_multi_arch_vpc_workspace
```

File: ci-operator/step-registry/upi/deprovision/powervs/cluster/upi-deprovision-powervs-cluster-commands.sh [hunk 6]
```diff
@@ -55,24 +55,6 @@ function ic() {
...
-
-function setup_ibmcloud_cli() {
-  if [ -z "$(command -v ibmcloud)" ]
-  then
-    echo "ibmcloud CLI doesn't exist, installing"
-    curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
-  fi
-
-  retry "ic config --check-version=false"
-  retry "ic version"
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/powervs/cluster/upi-deprovision-powervs-cluster-commands.sh:74]: We probably want the ic config --check-version=false and ic version just for debug issues.

File: ci-operator/step-registry/upi/deprovision/powervs/cluster/upi-deprovision-powervs-cluster-ref.yaml [hunk 7]
```diff
@@ -17,8 +17,10 @@ ref:
     default: ""
   - name: JQ_VERSION
     default: ""
+  - name: OCP_VERSION
+    default: "4.15"
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/powervs/cluster/upi-deprovision-powervs-cluster-ref.yaml:21]:
```suggestion
```
not in deprovision... only provisioning.

File: ci-operator/step-registry/upi/install/heterogeneous/vpc/upi-install-heterogeneous-vpc-commands.sh [hunk 8]
```diff
@@ -145,10 +133,10 @@ function get_ready_nodes_count() {
 function setup_mac_vpc_workspace(){
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/heterogeneous/vpc/upi-install-heterogeneous-vpc-commands.sh:133]:
```suggestion
function setup_multi_arch_vpc_workspace(){
```

File: ci-operator/step-registry/upi/install/heterogeneous/vpc/upi-install-heterogeneous-vpc-commands.sh [hunk 9]
```diff
@@ -194,14 +182,14 @@ EOF
 EOF
   cp "/tmp/powervs-config.json" "${SHARED_DIR}"/powervs-config.json
 
-  cp "${IBMCLOUD_HOME_FOLDER}"/ocp4-mac-vpc/var-mac-vpc.tfvars "${SHARED_DIR}"/var-mac-vpc.tfvars
-  cat "${IBMCLOUD_HOME_FOLDER}"/ocp4-mac-vpc/var-mac-vpc.tfvars
+  cp "${IBMCLOUD_HOME_FOLDER}"/ocp4-multi-arch-vpc/var-multi-arch-vpc.tfvars "${SHARED_DIR}"/var-multi-arch-vpc.tfvars
+  cat "${IBMCLOUD_HOME_FOLDER}"/ocp4-multi-arch-vpc/var-multi-arch-vpc.tfvars
 }
 
 function create_mac_vpc_resources() {
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/heterogeneous/vpc/upi-install-heterogeneous-vpc-commands.sh:189]:
```suggestion
function create_multi_arch_vpc_resources() {
```

File: ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh [hunk 10]
```diff
@@ -135,13 +121,13 @@ rhcos_import_image_filename     = "${COREOS_NAME}"
...
-cluster_id_prefix   = "rh-ci"
+cluster_id_prefix   = "multi-arch-p2"
 bastion   = { memory = "16", processors = "1", "count" = 1 }
 bootstrap = { memory = "16", processors = "0.5", "count" = 1 }
 master    = { memory = "16", processors = "0.5", "count" = 3 }
 worker    = { memory = "16", processors = "0.5", "count" = 2 }
-openshift_install_tarball = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OCP_VERSION}/ppc64le/openshift-install-linux.tar.gz"
-openshift_client_tarball  = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OCP_VERSION}/ppc64le/openshift-client-linux.tar.gz"
+openshift_install_tarball = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OPENSHIFT_VERSION}}/ppc64le/openshift-install-linux.tar.gz"
+openshift_client_tarball  = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OPENSHIFT_VERSION}}/ppc64le/openshift-client-linux.tar.gz"
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh:130]:
```suggestion
openshift_install_tarball = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OPENSHIFT_VERSION}/ppc64le/openshift-install-linux.tar.gz"
openshift_client_tarball  = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/${OCP_STREAM}/${OPENSHIFT_VERSION}/ppc64le/openshift-client-linux.tar.gz"
```

Comment (prb112) [reply ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh:130]: double }}

File: ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh [hunk 11]
```diff
@@ -153,17 +139,20 @@ ibm_cloud_cis_crn = "${IBMCLOUD_CIS_CRN}"
...
-  cp "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-mac-upi.tfvars "${SHARED_DIR}"/var-mac-upi.tfvars
-  cat "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-mac-upi.tfvars
+  cp "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-multi-arch-upi.tfvars "${SHARED_DIR}"/var-multi-arch-upi.tfvars
+  cat "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/var-multi-arch-upi.tfvars
 }
 
 function create_upi_powervs_cluster() {
   cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ || true
-  # Dev Note: https://github.com/ocp-power-automation/openshift-install-power/blob/devel/openshift-install-powervs#L767C1-L767C145
+  # Dev Note: https://github.com/ocp-pcower-automation/openshift-install-power/blob/devel/openshift-install-powervs#L767C1-L767C145
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh:148]:
```suggestion
  # Dev Note: https://github.com/ocp-power-automation/openshift-install-power/blob/devel/openshift-install-powervs#L767C1-L767C145
```

File: ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh [hunk 12]
```diff
@@ -416,33 +384,6 @@ function create_powervs_workspace() {
   done
 }
 
-function create_powervs_private_network() {
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh:419]: Where did this go? we still need it

File: ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-ref.yaml [hunk 13]
```diff
@@ -23,6 +23,11 @@ ref:
     default: ""
   - name: JQ_VERSION
     default: ""
+  - name: OCP_VERSION
+    default: "4.15"
+  - name: OPENSHIFT_VERSION
+    default: "4.15"
+    documentation: The version of OpenShift to install, in MAJOR.MINOR format.
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-ref.yaml:30]: One or the other... 
We should also update the job configuration so 4.15 is 4.15, 4.16 is 4.16 et cetra. 

see ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.16.yaml

**Review by prb112 — CHANGES_REQUESTED** [review]
left some comments

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh (hunk 3 above)
Comment (pkenchap) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:65]: Sure , I ad created the PR - https://github.com/IBM/ocp4-upi-compute-powervs-ibmcloud/pull/21 
As of now jira@ibm is down, once it up , will open an issue and will update the PR details there .

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.15.yaml [hunk 14]
```diff
@@ -165,20 +155,8 @@ tests:
...
-        debug\| oc idle\| Pods cannot access\| Image append should create\| Image
-        extract should extract\| Image info should display\| Image layer subresource\|
-        oc tag should change image\| when installed on the cluster should\| OpenShift
-        alerting rules\| The HAProxy router should\| egressrouter cni resources\|
-        pod should start\| pod sysctls\| Managed cluster should have same number of
-        Machines and Nodes \| Managed cluster should have machine resources \| should
-        not have pod creation failures during install \| DNS should answer queries
-        using the local DNS endpoint \| not repeat pathologically for ns/openshift-etcd
-        \| egressFirewall should have no impact outside its namespace
+      OPENSHIFT_VERSION: 4.15
```
Comment (prb112) [inline ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.15.yaml:158]: I think you have to insert this alphabetically before RESOURCE_GROUP

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.15.yaml [hunk 15]
```diff
@@ -141,17 +141,7 @@ tests:
...
-        extract should extract\| Image info should display\| Image layer subresource\|
-        oc tag should change image\| when installed on the cluster should\| OpenShift
-        alerting rules\| The HAProxy router should\| egressrouter cni resources\|
-        pod should start\| pod sysctls\| build volumes should mount given secrets
-        and configmaps into the build pod
+      TEST_SKIPS: deploymentconfigs\| should expose cluster services outside the cluster\| FIPS TestFIPS\| Multi-stage image builds should succeed\| Optimized image builds should succeed\| build can reference a cluster service\| custom build with buildah\| oc new-app should succeed\| prune builds based on settings\| s2i build with a root\| verify /run filesystem contents\| oc can run\| oc debug\| oc idle\| Pods cannot access\| Image append should create\| Image extract should extract\| Image info should display\| Image layer subresource\| oc tag should change image\| when installed on the cluster should\| OpenShift alerting rules\| The HAProxy router should\| egressrouter cni resources\| pod should start\| pod sysctls\| build volumes should mount given secrets and configmaps into the build pod
```
Comment (prb112) [inline ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.15.yaml:144]: Out of curiosity why did this change?

**Comment by pkenchap** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.15-ocp-e2e-powervs-ovn-multi-ppc64le

File: ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh (hunk 2 above)
Comment (prb112) [reply ci-operator/step-registry/upi/deprovision/heterogeneous/vpc/upi-deprovision-heterogeneous-vpc-commands.sh:85]: Pinged comments directly

**Comment by pkenchap** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.15-ocp-e2e-powervs-ovn-multi-ppc64le

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-ref.yaml [hunk 16]
```diff
@@ -21,5 +21,4 @@ ref:
     documentation: |-
       The is cucushift tag to name the workspace to mention its a cucushift test workflow.
   documentation: >-
-    This step deprovisions the powervs nodes created by upi-install-heterogeneous-powervs.
-    It should be executed before the ipi-deprovision step.
+    This step deprovisions the powervs nodes created by upi-install-heterogeneous-powervs. It should be executed before the ipi-deprovision step.
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-ref.yaml:24]: I wonder if the automation may fail on this one?

**Comment by prb112** [comment]
Let's restart in a new PR.

**Pull request closed without merging**